recipes-kernel/linux/files/patches-5.10/0119-arm64-dts-ti-iot2050-add-the-support-for-the-m.2-var.patch

```diff
@@ -0,0 +1,311 @@
+
+diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
+index e8a07d411627..3de28e26aa94 100644
+--- a/arch/arm64/boot/dts/ti/Makefile
++++ b/arch/arm64/boot/dts/ti/Makefile
+@@ -11,6 +11,10 @@ dtb-$(CONFIG_ARCH_K3) += k3-am6528-iot2050-basic.dtb
+ dtb-$(CONFIG_ARCH_K3) += k3-am6528-iot2050-basic-pg2.dtb
+ dtb-$(CONFIG_ARCH_K3) += k3-am6548-iot2050-advanced.dtb
+ dtb-$(CONFIG_ARCH_K3) += k3-am6548-iot2050-advanced-pg2.dtb
++dtb-$(CONFIG_ARCH_K3) += k3-am6548-iot2050-advanced-m2.dtb
++dtb-$(CONFIG_ARCH_K3) += k3-am654-iot2050-bkey-pciex2.dtbo
++dtb-$(CONFIG_ARCH_K3) += k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo
++dtb-$(CONFIG_ARCH_K3) += k3-am654-iot2050-bkey-pcie-ekey-pcie.dtbo
+ 
+ dtb-$(CONFIG_ARCH_K3) += k3-j721e-common-proc-board.dtb
+ 
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi
+new file mode 100644
+index 000000000000..40ad18747d95
+--- /dev/null
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2-m2.dtsi
+@@ -0,0 +1,33 @@
++// SPDX-License-Identifier: GPL-2.0
++/*
++ * Copyright (c) Siemens AG, 2021
++ *
++ * Authors:
++ *   Chao Zeng <chao.zeng@siemens.com>
++ *   Jan Kiszka <jan.kiszka@siemens.com>
++ *
++ * Common bits of the IOT2050 Basic and Advanced variants, PG2
++ */
++
++&main_pmx0 {
++	cp2102n_reset_pin_default: cp2102n-reset-pin-default {
++		pinctrl-single,pins = <
++			/* (AF12) GPIO1_24, used as cp2102 reset */
++			AM65X_IOPAD(0x01e0, PIN_OUTPUT, 7)
++		>;
++	};
++};
++
++&main_gpio1 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&cp2102n_reset_pin_default>;
++	gpio-line-names =
++		"", "", "", "", "", "", "", "", "", "",
++		"", "", "", "", "", "", "", "", "", "",
++		"", "", "", "", "CP2102N-RESET";
++};
++
++&dss {
++	/* Workaround needed to get DP clock of 154Mhz */
++	assigned-clocks = <&k3_clks 67 0>;
++};
+\ No newline at end of file
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2.dtsi
+index e73458ca6900..48f9242496cb 100644
+--- a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2.dtsi
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common-pg2.dtsi
+@@ -9,28 +9,7 @@
+  * Common bits of the IOT2050 Basic and Advanced variants, PG2
+  */
+ 
+-&main_pmx0 {
+-	cp2102n_reset_pin_default: cp2102n-reset-pin-default {
+-		pinctrl-single,pins = <
+-			/* (AF12) GPIO1_24, used as cp2102 reset */
+-			AM65X_IOPAD(0x01e0, PIN_OUTPUT, 7)
+-		>;
+-	};
+-};
+-
+-&main_gpio1 {
+-	pinctrl-names = "default";
+-	pinctrl-0 = <&cp2102n_reset_pin_default>;
+-	gpio-line-names =
+-		"", "", "", "", "", "", "", "", "", "",
+-		"", "", "", "", "", "", "", "", "", "",
+-		"", "", "", "", "CP2102N-RESET";
+-};
+-
+-&dss {
+-	/* Workaround needed to get DP clock of 154Mhz */
+-	assigned-clocks = <&k3_clks 67 0>;
+-};
++#include "k3-am65-iot2050-common-pg2-m2.dtsi"
+ 
+ &serdes0 {
+ 	assigned-clocks = <&k3_clks 153 4>, <&serdes0 AM654_SERDES_CMU_REFCLK>;
```

Comment:
What about the dwc3_0 and usb0 nodes? Truly different on M2?

Reply:
yes, usb3.0 connects to the M.2 connector.
there is no USB3.0 ports on the interface,two ports only support usb2.0.
this is different with pg2. So I abstract one dtsi header file for m2 and pg2.

I think it is necessary to discuss whether this method is reasonable. Do you have any better suggestions?